wom/src/main/scala/wom/callable/TaskDefinition.scala

```diff
@@ -5,84 +5,89 @@ import common.validation.ErrorOr.ErrorOr
+                                        commandPartSeparator: String = "",
+                                        stdoutRedirection: Option[String] = None,
+                                        stderrRedirection: Option[String] = None
+                                       ) extends TaskDefinition
 
+/**
+  * A task definition with an embedded graph.
+  * Can be called from a workflow but can also be run as a standalone execution.
+  */
+final case class ExecutableTaskDefinition private (callableTaskDefinition: CallableTaskDefinition,
+                                                   override val graph: Graph
+                                                  ) extends TaskDefinition with ExecutableCallable {
+  override def name = callableTaskDefinition.name
+  override def inputs = callableTaskDefinition.inputs
+  override def outputs = callableTaskDefinition.outputs
+
+  override def commandTemplate = callableTaskDefinition.commandTemplate
+  override def runtimeAttributes = callableTaskDefinition.runtimeAttributes
+  override def meta = callableTaskDefinition.meta
+  override def parameterMeta = callableTaskDefinition.parameterMeta
+  override def prefixSeparator = callableTaskDefinition.prefixSeparator
+  override def commandPartSeparator = callableTaskDefinition.commandPartSeparator
+  override def stdoutRedirection = callableTaskDefinition.stdoutRedirection
+  override def stderrRedirection = callableTaskDefinition.stderrRedirection
+}
 
-  // TODO: fixup? The general version in Callable might not be good enough for Task:
-//  def evaluateOutputs(inputs: EvaluatedTaskInputs,
-//                      wdlFunctions: WdlStandardLibraryFunctions,
-//                      postMapper: WomValue => Try[WomValue] = v => Success(v)): Try[Map[TaskOutput, WomValue]] = {
-//    val fqnInputs = inputs map { case (d, v) => d.fullyQualifiedName -> v }
-//    val evaluatedOutputs = outputs.foldLeft(Map.empty[TaskOutput, Try[WomValue]])((outputMap, output) => {
-//      val currentOutputs = outputMap collect {
-//        case (outputName, value) if value.isSuccess => outputName.fullyQualifiedName -> value.get
-//      }
-//      def knownValues = currentOutputs ++ fqnInputs
-//      val lookup = NoLookup // TODO: Reinstate: val lookup = lookupFunction(knownValues, wdlFunctions, relativeTo = output)
-//      val coerced = output.expression.evaluate(lookup, wdlFunctions) flatMap output.womType.coerceRawValue
-//      val jobOutput = output -> (coerced flatMap postMapper).recoverWith {
-//        case t: Throwable => Failure(new RuntimeException(s"Could not evaluate ${output.name} = ${output.expression.toWdlString}", t))
-//      }
-//      outputMap + jobOutput
-//    }) map { case (k, v) => k -> v }
-//
-//    TryUtil.sequenceMap(evaluatedOutputs, "Failed to evaluate outputs.")
-//  }
+object ExecutableTaskDefinition {
+  def tryApply(callableTaskDefinition: CallableTaskDefinition): ErrorOr[ExecutableTaskDefinition] =
```

Comment:
ToL I'd prefer the name `apply` here but I predict you are reserving that for a constructor that returns the value w/ no effect.